From be048baed53ffe1cb45065f2d860e7a7441f4092 Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Mon, 10 Sep 2018 17:12:49 +0300
Subject: [PATCH 295/706] enetc: replace ioread64 for 32b systems

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
(cherry picked from commit 5c2d35a9aa4762477674dc3d766c34998e15f3d7)
(cherry picked from commit f5447233e3f21e830ae3f862f37fd01c26a996c0)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_hw.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_hw.h b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
index cf0486ea0230..1b11981fa97f 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_hw.h
+++ b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
@@ -338,7 +338,15 @@ struct enetc_hw {
 #define enetc_wr_reg(reg, val)	iowrite32((val), (reg))
 #define enetc_rd(hw, off)	enetc_rd_reg((hw)->reg + (off))
 #define enetc_wr(hw, off, val)	enetc_wr_reg((hw)->reg + (off), val)
+#ifdef ioread64
 #define enetc_rd_reg64(reg)		ioread64((reg))
+#else
+/* using this to read out stats on 32b systems */
+static inline u64 enetc_rd_reg64(void __iomem *reg)
+{
+	return __le64_to_cpu(*(u64*)reg);
+}
+#endif
 #define enetc_wr_reg64(reg)		iowrite64((val), (reg))
 #define enetc_rd64(hw, off)		enetc_rd_reg64((hw)->reg + (off))
 #define enetc_wr64(hw, off, val)	enetc_wr_reg64((hw)->reg + (off), val)
-- 
2.17.1

